### ISSUE TITLE
Unable to splint right arm/leg or swap side of splints

### ISSUE THREAD

**Issue by Wyghab** [CleverRaven/Cataclysm-DDA#56721]
### Describe the bug

With a broken right arm, equipping an arm splint turns the arm splint into arm splint(left). The game does not equip the arm splint with the message "there is no limb that this would benefit". The arm cannot be changed to arm splint(right) unless worn even when attempting to chance side through the inventory interface.

Additionally, breaking both arms, and attempting to swap side of an existing splint generates a message saying that "your splint conflicts with hard armor on your other side, so you can't switch it" even if naked. Same story if tried from inventory.

Leg splints behave the same way, with the same issues.

### Steps to reproduce

1. Make a player, set Right arm health to 0, spawn an arm splint.
2. Equip arm splint, observe that you get the message "no limb would benefit from this". The arm splint has turned into arm splint(left).
3. Go into inventory and details of the splint, press c to attempt to chance side. Produces message a saying that it has to be equipped to change side.
4. Unequip all clothes, set left arm health to 0, then  attempt to change side through inventory, or equip it and use clothing layering to do the same. Produces a message about hard armor on other side interfering.
5. Repeat same steps for Legs instead of arms, observe identical behaviour.

### Expected behavior

Equipping a splint should put it on the correct side, or at least give you the option of which side it should be placed on. Alternatively it should be able to be side-swapped without being equipped as it currently can't be equipped.

### Screenshots

![image](https://user-images.githubusercontent.com/47194578/162602538-c4746def-a2e7-45f2-8cbd-7f91da59e677.png)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19043.1586 (21H1)
- Game Version: b54ab40 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

**Comment by Wyghab** [comment]
I'd like to add that, since I just tried it, but breaking both limbs and equipping two unassigned splints does splint the right side on the 2nd one. Which would lead me to believe that the logic for which limb to splint simply doesn't consider the right side if the left side doesn't have something rigid taking that spot already.

**Comment by YesCarNoCar** [comment]
I have this issue too and can add that after I put on something rigid on your left arm (scrap metal arm guards) it still does not allow  me to put on either a arm splint (left) or an unassigned one. The message on the former being:
arm splint (left) "You don't have any broken limbs this could help"
and the second one being:
arm splint "Can't wear more than one rigid item on left forearm"

**Comment by Wyghab** [comment]
I've been doing a little digging since no ones really taken notice of this issue, not able to investigate it properly since I've yet to get compilation of the source. 

So far I've narrowed it down to the function Character::can_wear in character_attire. 
The variable need_splint does not get assigned correctly for when only the right side of broken.
### character_attire.cpp line ~94
```
if( it.has_flag( flag_SPLINT ) ) {
  bool need_splint = false;
  for( const bodypart_id &bp : get_all_body_parts() ) {
    if( !it.covers( bp ) ) {
      continue;
    }
    if( is_limb_broken( bp ) && !worn_with_flag( flag_SPLINT, bp ) ) {
      need_splint = true;
      break;
    }
  }
  // need_splint gets improperly assigned when right side is broken, & left side is not splinted.
  if( !need_splint ) {
    return ret_val<bool>::make_failure( is_avatar() ?
	_( "You don't have any broken limbs this could help." )
	: _( "%s doesn't have any broken limbs this could help." ), get_name() );
  }
}
```
And I'm pretty suspicious of the covers function, since to me it seems like it's the most likely one to fail, if it's not cover, it's "is_limb_broken" or "worn_with_flag", but I didn't see anything that particularly stood out at a glance.
### item.cpp line ~813
```
bool item::covers( const bodypart_id &bp ) const
{
  bool does_cover = false;
  // Could get_side() return improper value? Causing only the first option (left) to be considered?
  iterate_covered_body_parts_internal( get_side(), [&]( const bodypart_str_id & covered ) {
    // Alternatively... does comparing pointer addresses make sense in bp == covered?
    does_cover = does_cover || bp == covered;
  } );
  return does_cover;
}
```

**Comment by Wyghab** [comment]
> I have this issue too and can add that after I put on something rigid on your left arm (scrap metal arm guards) it still does not allow me to put on either a arm splint (left) or an unassigned one. The message on the former being: arm splint (left) "You don't have any broken limbs this could help" and the second one being: arm splint "Can't wear more than one rigid item on left forearm"

Well, that's interesting. Guess it might pay to look closer into worn_with_flag, since a splinted left arm allows splinting right arm. But rigid item doesn't allow consideration of right arm.

**Comment by Wyghab** [comment]
Besides that, is there an actual reason why you shouldn't be able to equip splints with completely fine, unbroken arms? Understandable they are crap in terms of armor and encumberance, but you should still be able to wear one.

**Comment by Wyghab** [comment]
Okay I believe I have 100% cornered this bug now.

Inside character::wear_item, can_wear is called on a copy of the splint to be worn that has it's side already set to LEFT, however it is still possible at this point that it should be RIGHT, since it never checked for which side to set a splint to. The check for which side to place a splint is handled in Item::worn, hence it is never triggered since the splint is never actually put on!

**Comment by Wyghab** [comment]
The code that causes the issue is the code for checking rigid sidedness, since it sets the side of the splint to LEFT
## character_attire.cpp line ~226
```
worn.check_rigid_sidedness( to_wear_copy );
worn.one_per_layer_sidedness( to_wear_copy );

auto result = wear_item( to_wear_copy, interactive );
```
Then when the item passes into wear_item, there is another check for can_wear, which this time fails because the item can now only fit on left side.
## line ~320
```
const auto ret = can_wear( to_wear );
if( !ret.success() ) {
  if( interactive ) {
    add_msg_if_player( m_info, "%s", ret.c_str() );
  }
  return cata::nullopt;
}
```
Now, there is code to properly set splint side inside item::on_wear, but this code is not triggered unless the item is actually worn. Ghetto fix would be simply to set items with flag_SPLINT to side::BOTH, but it might be a deeper problem than just splints due to how item::on_wear already sets side on items based on lowest encumberance.
## Ghetto Solution
```
worn.check_rigid_sidedness( to_wear_copy );
worn.one_per_layer_sidedness( to_wear_copy );
if ( to_wear_copy .has_flag( flag_SPLINT ) ) {
    to_wear_copy.set_side( side::BOTH );
}
auto result = wear_item( to_wear_copy, interactive );
```
Can confirm that the ghetto solution allows splinting of right side legs&arms due to on_wear correctly placing splints where they should be.
![image](https://user-images.githubusercontent.com/47194578/166308451-7c1d8745-8bf6-4007-a536-a53a00a3423e.png)

**Comment by Wyghab** [comment]
I've narrowed down the issue with swap side, the issue is that logic is performed on the equipped item, instead of a copy. 
![image](https://user-images.githubusercontent.com/47194578/167163535-d1e2b790-98cd-494d-8ccc-97d622c2f407.png)
The worn item is side swapped, then the code checks for rigid conflicts on the bodyparts the item occupies, which is guaranteed to fail since the item itself occupies that space.
I played around with simply making a copy item for the comparison checks, but met with some weird bugs, so it's not super simple to fix. But it did allow swapping sides of splints after using a copy that is not worn for the "which bodyparts to check"
